omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,137 @@
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	public void shouldStartTestTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldStopTestTask() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask.getName());
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTestTaskAlreadyStarted() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask.getName());
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask.getName()));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTestTaskAlreadyStopped() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[" + testTaskMockJson + "]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask.getName())));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+	}
+	
+	@Test
+	public void shouldUnloadTestTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask.getName()));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"delete\", \"tasks\":[" + testTaskMockJson + "]}")));
+		assertThat(mockTaskServiceWrapper.registeredTasks, not(hasItem(testTask.getName())));
+	}
+	
+	@Override
+	@Test(expected = Exception.class)
+	public void shouldGetDefaultByUuid() throws Exception {
+		super.shouldGetDefaultByUuid();
+	}
+	
+	@Override
+	@Test(expected = Exception.class)
+	public void shouldGetRefByUuid() throws Exception {
+		super.shouldGetRefByUuid();
+	}
+	
+	@Override
+	@Test(expected = Exception.class)
+	public void shouldGetFullByUuid() throws Exception {
+		super.shouldGetFullByUuid();
+	}
+	
+	@Override
+	@Test(expected = Exception.class)
+	public void shouldGetAll() throws Exception {
+		super.shouldGetAll();
```

Comment:
Does this mean that we do not support getting all tasks?

Reply:
yah It will not support to that

Reply:
Oh you are right, since these are just task actions. 😊